syftlib/src/main/java/org/openmined/syft/processes/SyftJob.kt

```diff
@@ -0,0 +1,172 @@
+    fun setRequestKey(responseData: CycleResponseData.CycleAccept) {
+        requestKey = responseData.requestKey
+        cycleStatus.set(CycleStatus.ACCEPTED)
+        jobStatusProcessor.offer(JobStatusMessage.JobCycleAccepted)
+    }
+
+    fun downloadData(destinationDir: String) {
+        trainingParamsStatus.set(DownloadStatus.RUNNING)
+        plans.forEach { (planId, plan) -> downloadPlanFile("$destinationDir/plans", planId, plan) }
+
+        protocols.forEach { (protocolId, protocol) ->
+            downloadProtocolFile(
+                "$destinationDir/protocols",
+                protocolId,
+                protocol
+            )
+        }
+        downloadModelFile(destinationDir, modelName)
+        trainingParamsStatus.set(DownloadStatus.COMPLETE)
+    }
+
+    fun downloadModelFile(destinationDir: String, modelName: String) {
+        compositeDisposable.add(
+            worker.getDownloader().downloadModel(worker.workerId, requestKey, modelName).compose(
+                schedulers.applySingleSchedulers()
+            ).flatMap { response ->
+                saveFile(
+                    response.body()?.byteStream(),
+                    destinationDir,
+                    modelName
+                )
+            }.subscribe { fileLocation: String -> modelFile = fileLocation })
+    }
+
+    fun downloadPlanFile(destinationDir: String, planId: String, plan: Plan) {
+        compositeDisposable.add(worker.getDownloader().downloadPlan(
+            worker.workerId,
+            requestKey,
+            planId,
+            "torchscript"
+        ).compose(schedulers.applySingleSchedulers())
+                .flatMap { response ->
+                    saveFile(
+                        response.body()?.byteStream(),
+                        destinationDir,
+                        planId
+                    )
+                }.subscribe { fileLocation: String -> plan.torchScriptLocation = fileLocation })
+    }
+
+    fun downloadProtocolFile(destinationDir: String, protocolId: String, protocol: Protocol) {
+        compositeDisposable.add(worker.getDownloader().downloadProtocol(
+            worker.workerId,
+            requestKey,
+            protocolId
+        ).compose(schedulers.applySingleSchedulers())
+                .flatMap { response ->
+                    saveFile(
+                        response.body()?.byteStream(),
+                        destinationDir,
+                        protocolId
+                    )
+                }.subscribe { fileLocation: String ->
+                    protocol.protocolFileLocation = fileLocation
+                })
+    }
+
+    private fun saveFile(
+        input: InputStream?,
+        destinationDir: String,
+        fileName: String
+    ): Single<String> {
+        val destination = File(destinationDir)
+        if (!destination.exists())
+            destination.mkdirs()
+        return Single.create { emitter ->
+            if (input == null)
```

Comment:
I prefer to use `input?.let{...} ?: emitter.onError(...)`
(it's a personal preference though)